app/controllers/api/v0/users_controller.rb

```diff
@@ -62,9 +62,13 @@ def token
 
   private
 
-    def show_user(user, show_rankings: false)
+    def show_user(user, show_rankings: false, current_user: false)
       if user
-        json = { user: user }
+        if current_user
+          json = { user: user.serializable_hash(private_attributes: ['email']) }
+        else
+          json = { user: user }
+        end
```

Comment:
This if-clause looks pretty repetitive, but I don't have an immediate better replacement in mind.
What do you think about passing in desired private user attributes as an array parameter directly into the method? That way, we avoid the naming discussion from the other comment.

Unsure whether that would be more readable, but it may be worth a try.

Reply:
Oh that's better actually. I've done that now. Thanks.